[fluidsynth] Update patch and portfile

FtZPetruska opened the pull request:
- [X] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)
- [X] SHA512s are updated for each updated download
- [X] The "supports" clause reflects platforms that may be fixed by this new version
- [X] Any fixed [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt) entries are removed from that file.
- [X] Any patches that are no longer applied are deleted from the port's directory.
- [X] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [X] Only one version is added to each modified port's versions file.

---

The updated `gentables.patch` from #29038 accidentally disables the installation of `FluidSynthTargets.cmake`.

Additionally, this PR cleans up a few things in the portfile:
- Install a usage file
- Remove the manual cleanup of `bin` directories since `vcpkg_copy_tools(AUTO_CLEAN)` already handles it
- Copy pdbs
- Remove the installed manpages
- Remove the `enable-debug` option since it no longer exists upstream

Tested with default features on:
- Windows (x64-windows, x86-windows, x64-windows-static)
- macOS (arm64-osx, arm64-osx-dynamic)
- Linux (x64-linux)

Tested with all features on:
- Windows (x64-windows, x86-windows)
- macOS (arm64-osx)

FtZPetruska committed:
[fluidsynth] Update gentables patch.

Installation of FluidSynthTargets.cmake was accidentally turned off with
the previous version of the patch.

FtZPetruska committed:
[fluidsynth] Update portfile.

- Remove manpages.
- vcpkg_copy_tools already cleans up the bin directories
- Copy pdbs after moving the tools.
- Remove `enable-debug` as it no longer exists.

FtZPetruska committed:
[fluidsynth] Install usage.

FtZPetruska committed:
[fluidsynth] Bump port version.

FtZPetruska committed:
[fluidsynth] Update baseline.

FtZPetruska committed:
[fluidsynth] Find dependencies in the config file.

The library dependencies of `FluidSynth::libfluidsynth-OBJ` were not
searched for in the Config file.

FtZPetruska committed:
[fluidsynth] Update version.

jimwang118 commented on ports/fluidsynth/gentables.patch:
```diff
@@ -1,59 +1,52 @@
+ 
+ # ******* Auto Generated Lookup Tables ******
  
 +option(VCPKG_BUILD_MAKE_TABLES "Build `make_tables`" OFF)
```
```suggestion

```
It is recommended to modify the value of VCPKG_BUILD_MAKE_TABLES in the option of portfile.cmake.

FtZPetruska committed:
[fluidsynth] Apply suggestion from review.

Do not set `VCPKG_BUILD_MAKE_TABLES` in the patch.

FtZPetruska committed:
[fluidsynth] Update version.

FtZPetruska commented on ports/fluidsynth/add-usage-requirements.patch:
```diff
@@ -0,0 +1,17 @@
+ 
+ # finally, include the targets file
++include(CMakeFindDependencyMacro)
++find_dependency(OpenMP COMPONENTS C REQUIRED)
++find_dependency(PkgConfig REQUIRED)
```
Should I make it `QUIET` instead of `REQUIRED` to avoid polluting stdout?

dg0yt replied:
Just rely on `find_dependency` doing the right thing with regard to both `QUIET` and `REQUIRED`. These two options are the core compentence of that macro.

FtZPetruska committed:
[fluidsynth] Update patch.

- Avoid using REQUIRED to find packages.
- Check for ALSA on Linux.
- Provide versions for pkg-config modules.

FtZPetruska committed:
[fluidsynth] Update version.

jimwang118 commented:
All test passed with following triplets:

x86-windows
x64-windows
x64-windows-static
